### PR TITLE
fix(datasource/go): support go proxy with abortOnError

### PR DESCRIPTION
**Pull request by ahippler**
## Changes

Do not fail goproxy version lookup with `abortOnError: true`.

## Context

The goproxy registry api only support listing versions per major version. Renovate tries to fetch the versions for the next major versions to be able to detect major version updated. This is repeated until the major version does not exist and the last requests fails with a 404. With abortOnError enabled the whole lookup throws an error for the last request. The logic to ignore the error does not handle the wrapper error.

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/modules/datasource/go/releases-goproxy.spec.ts [hunk 1]
```diff
@@ -195,46 +205,50 @@ describe('modules/datasource/go/releases-goproxy', () => {
...
-        .get('/@v/list')
-        .replyWithError('unknown');
-
-      httpMock
-        .scope(`${baseUrl}/github.com/google/btree`)
-        .get('/@v/list')
-        .reply(
-          200,
-          codeBlock`
+    it.each([true, false])(
```
Comment (viceice) [inline lib/modules/datasource/go/releases-goproxy.spec.ts:208]: please use string template syntax

File: lib/modules/datasource/go/releases-goproxy.spec.ts [hunk 2]
```diff
@@ -406,53 +420,57 @@ describe('modules/datasource/go/releases-goproxy', () => {
...
-    it('handles major releases', async () => {
-      process.env.GOPROXY = baseUrl;
-
-      httpMock
-        .scope(`${baseUrl}/github.com/google/btree`)
-        .get('/@v/list')
-        .reply(
-          200,
-          codeBlock`
+    it.each([true, false])(
```
Comment (viceice) [inline lib/modules/datasource/go/releases-goproxy.spec.ts:423]: same here

File: lib/modules/datasource/go/releases-goproxy.spec.ts (hunk 1 above)
Comment (ahippler) [reply lib/modules/datasource/go/releases-goproxy.spec.ts]: maybe its better readable with multiple columns but you loose the type support

File: lib/modules/datasource/go/releases-goproxy.spec.ts [hunk 3]
```diff
@@ -195,46 +205,54 @@ describe('modules/datasource/go/releases-goproxy', () => {
...
-        .get('/@v/list')
-        .replyWithError('unknown');
-
-      httpMock
-        .scope(`${baseUrl}/github.com/google/btree`)
-        .get('/@v/list')
-        .reply(
-          200,
-          codeBlock`
+    it.each`
```
Comment (viceice) [inline lib/modules/datasource/go/releases-goproxy.spec.ts:208]: you can still have type support

https://github.com/renovatebot/renovate/blob/d65effaa77b53ef93312474e81fdbd0efeff0498/lib/modules/datasource/nuget/common.spec.ts#L4

**Review by viceice — APPROVED** [review]

**Pull request merged**